javascript-carousel/main.js

```diff
@@ -0,0 +1,77 @@
+var $circle = document.querySelectorAll('.circle');
+
+$carousel.addEventListener('click', function () {
+  clearInterval(advanceIntervalId);
+
+  if (event.target === $chevLeft) {
+    for (var i = 0; i < $image.length; i++) {
+      if ($image[i].getAttribute('class') === 'image') {
+        $image[i].setAttribute('class', 'image hidden');
+        $circle[i].setAttribute('class', 'far fa-circle selector circle');
+        if (i === 0) {
+          $image[$image.length - 1].setAttribute('class', 'image');
+          $circle[$circle.length - 1].setAttribute('class', 'fas fa-circle selector circle');
+          break;
+        } else {
+          $image[i - 1].setAttribute('class', 'image');
+          $circle[i - 1].setAttribute('class', 'fas fa-circle selector circle');
+        }
+      }
+    }
+  } else if (event.target === $chevRight) {
+    for (var j = 0; j < $image.length; j++) {
+      if ($image[j].getAttribute('class') === 'image') {
+        $image[j].setAttribute('class', 'image hidden');
+        $circle[j].setAttribute('class', 'far fa-circle selector circle');
+        if (j === $image.length - 1) {
+          $image[0].setAttribute('class', 'image');
+          $circle[0].setAttribute('class', 'fas fa-circle selector circle');
+        } else {
+          $image[j + 1].setAttribute('class', 'image');
+          $circle[j + 1].setAttribute('class', 'fas fa-circle selector circle');
+          break;
+        }
+      }
+    }
+  } else if (event.target.getAttribute('class') === 'far fa-circle selector circle') {
+    for (var k = 0; k < $circle.length; k++) {
+      if ($circle[k].getAttribute('class') === 'fas fa-circle selector circle') {
+        $circle[k].setAttribute('class', 'far fa-circle selector circle');
+        $image[k].setAttribute('class', 'image hidden');
+      }
+    }
+    for (var l = 0; l < $circle.length; l++) {
+      if (event.target.getAttribute('id') === $circle[l].getAttribute('id')) {
+        $image[l].setAttribute('class', 'image');
+      }
+      event.target.setAttribute('class', 'fas fa-circle selector circle');
+    }
+  }
+  advanceIntervalId = setInterval(advance, 3000);
+});
+
+var advanceIntervalId = setInterval(advance, 3000);
+
+function advance() {
+  for (var m = 0; m < $image.length; m++) {
+    if (m !== $image.length - 1) {
+      if ($image[m].getAttribute('class') === 'image') {
+        $image[m].setAttribute('class', 'image hidden');
+        $circle[m].setAttribute('class', 'far fa-circle selector circle');
+        $image[m + 1].setAttribute('class', 'image');
+        $circle[m + 1].setAttribute('class', 'fas fa-circle selector circle');
+        break;
+      }
+    } else {
+      $image[m].setAttribute('class', 'image hidden');
+      $circle[m].setAttribute('class', 'far fa-circle selector circle');
+      $image[0].setAttribute('class', 'image');
+      $circle[0].setAttribute('class', 'fas fa-circle selector circle');
+      break;
+    }
+  }
```

Comment:
If you make the above changes suggested and put that logic in a function, this code could just become:
1. pick new active index
1. run the logic described above